pkg/kdump/kdump-view.jsx

```diff
@@ -276,9 +271,33 @@ class KdumpPage extends React.Component {
                     // we can get rid of the entry altogether
                     delete settings["core_collector"];
                 }
-            } else {
-                console.log("not changing compression setting");
-                return;
+            }
+        } else if (key === "target") {
+            /* target changed, restore settings and wipe all settings associated
+             * with a target so no conflicting settings remain */
+            settings = {};
+            Object.keys(this.props.kdumpStatus.config).forEach((key) => {
+                settings[key] = cockpit.extend({}, this.props.kdumpStatus.config[key]);
+            });
+            Object.keys(this.props.kdumpStatus.target).forEach((key) => {
+                if (settings[key])
+                    delete settings[key];
+            });
+            if (value === "ssh")
+                settings.ssh = { value: "" };
+            else if (value === "nfs")
+                settings.nfs = { value: "" };
+
+            if ("core_collector" in settings &&
+                settings["core_collector"].value.includes("makedumpfile")) {
+                if (value === "ssh" && !settings["core_collector"].value.includes("-F"))
+                    settings["core_collector"].value += " -F";
+                else if (settings["core_collector"].value.includes("-F"))
+                    settings["core_collector"].value =
+                        settings["core_collector"].value
+                                .split(" ")
+                                .filter((e) => { return (e != "-F") })
```

Comment:
Not important, just FYI: Let's avoid single-line blocks in arrows. This is better written as 

    .filter(e => e  != "-F")

This whole block could also use a comment  about the purpose, i. e. what it does with moving around the `-F` option.